dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImplIntegrationTest.java

```diff
@@ -0,0 +1,309 @@
+        final Map<String, Object> hostMap = host.getMap();
+        final String newHostname = "newHostName_" + System.currentTimeMillis();
+        hostMap.put("text1", newHostname);
+        hostMap.put("contentletInode", hostMap.get("inode"));
+
+        contentletWebAPI.saveContent(hostMap, false, false, user);
+        waitUntilJobIsFinish();
+
+        final Host hostFromDataBse = APILocator.getHostAPI().find(host.getIdentifier(), user, false);
+        assertEquals(newHostname, hostFromDataBse.getHostname());
+
+        checkTemplate(oldVersionInode, user, host.getHostname(), newHostname);
+        checkTemplate(liveVersionInode, user, newHostname, host.getHostname());
+        checkTemplate(workingVersionInode, user, newHostname, host.getHostname());
+
+        final TemplateLayout templateLayout = DotTemplateTool.themeLayout(template.getInode());
+        final String drawedBodyJson = JsonTransformer.mapper.writeValueAsString(templateLayout);
+        assertFalse(drawedBodyJson.contains(host.getHostname()));
+        assertTrue(drawedBodyJson.contains(newHostname));
+
+        final Template workingVersion = (Template) APILocator.getVersionableAPI().findWorkingVersion(
+                template.getIdentifier(), APILocator.systemUser(), false);
+
+        assertFalse(workingVersion.getDrawedBody().contains(host.getHostname()));
+        assertTrue(workingVersion.getDrawedBody().contains(newHostname));
+
+        final FileAssetContainer containerFromDataBase = (FileAssetContainer) APILocator.getContainerAPI().getWorkingContainerByFolderPath(
+                FileAssetContainerUtil.getInstance().getFullPath(hostFromDataBse, ((FileAssetContainer) container).getPath()),
+                user, false,
+                null);
+
+        assertEquals(newHostname, containerFromDataBase.getHost().getName());
+    }
+
+    private void checkTemplate(
+            final String templateInode,
+            final User user,
+            final String hostInTemplate,
+            final String hostNotInTemplate) throws DotSecurityException, DotDataException, JsonProcessingException {
+
+        final Template templateFromDatabase = APILocator.getTemplateAPI().find(templateInode, user, false);
+
+        assertFalse(templateFromDatabase.getDrawedBody().contains(hostNotInTemplate));
+        assertFalse(templateFromDatabase.getBody().contains(hostNotInTemplate));
+
+        assertTrue(templateFromDatabase.getDrawedBody().contains(hostInTemplate));
+        assertTrue(templateFromDatabase.getBody().contains(hostInTemplate));
+    }
+
+
+    /**
+     * Method to Test: {@link ContentletWebAPIImpl#saveContent(Map, boolean, boolean, User)}
+     * When: Change a Host' name and exists any html template layout using this
+     * Should: Update the container path into the template
+     *
+     * */
+    @Test
+    public void whenHostNameChangeWithLegacyTemplateLayout() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5990100)